src/opcodes/opcodes.cairo

```diff
@@ -174,6 +194,27 @@ pub mod Opcode {
         engine.dstack.push_int(a + b);
     }
 
+    fn opcode_greaterthan(ref engine: Engine) {
+        let a = engine.dstack.pop_int();
+        let b = engine.dstack.pop_int();
+        engine.dstack.push_int(if a > b {
+            0
+        } else {
+            1
+        });
+    }
+
+    fn opcode_within(ref engine: Engine) {
```

Comment:
Please refer to this implementation to see the proper ordering of variables on the stack : https://github.com/btcsuite/btcd/blob/b161cd6a199b4e35acec66afc5aad221f05fe1e3/txscript/opcode.go#L1844

Reply:
Working on this

Reply:
Hi, opcode_within is number 165 in the order, please could you tell me what you mean? I have pushed an update where I renamed and reordered the variables though

Reply:
So we have to make sure the values poped off the stack are in the same order as the bitcoin spec, so here `max_value` should be popped first, then `min_value`, then `selected_value`

Reply:
<img width="573" alt="Screenshot 2024-07-26 at 17 06 37" src="https://github.com/user-attachments/assets/2c730073-ce96-4a7a-ad03-847984a47494">

Hi @b-j-roberts I have rearranged the variables based on the order max before min before value